src/adapter/graphql/src/mutations/flows_mut/flows_mut_utils.rs

```diff
@@ -128,6 +129,46 @@ pub(crate) async fn ensure_flow_preconditions(
             };
         }
         DatasetFlowType::HardCompaction => (),
+        DatasetFlowType::Reset => {
+            if let Some(flow_configuration) = flow_run_configuration
+                && let FlowRunConfiguration::Reset(reset_configuration) = flow_configuration
+            {
+                let dataset_repo = from_catalog::<dyn kamu_core::DatasetRepository>(ctx).unwrap();
+
+                let dataset = dataset_repo
+                    .get_dataset(&dataset_handle.as_local_ref())
+                    .await
+                    .int_err()?;
+                let new_head_block_result = dataset
+                    .as_metadata_chain()
+                    .get_block(&reset_configuration.new_head_hash)
+                    .await;
```

Comment:
We do not actually need to read the block to check if it exists. Not a large read, but still.

There is such a method:

```
#[async_trait]
pub trait MetadataBlockRepository: Send + Sync {
    async fn contains_block(&self, hash: &Multihash) -> Result<bool, ContainsBlockError>;
    ...
}
```

We might need to bring it up through `MetadataChain` interface.